core/src/main/java/org/opensearch/sql/utils/IPUtils.java

```diff
@@ -0,0 +1,33 @@
+package org.opensearch.sql.utils;
+
+import com.google.common.net.InetAddresses;
+import org.opensearch.sql.data.model.ExprBooleanValue;
+import org.opensearch.sql.data.model.ExprValue;
+
+import java.util.Arrays;
+
+public class IPUtils {
+
+    /**
+     * Returns whether the given IP address is within the specified IP address range.
+     * Supports both IPv4 and IPv6 addresses.
+     *
+     * @param addressExprValue IP address (e.g. "198.51.100.14" or "2001:0db8::ff00:42:8329").
+     * @param rangeExprValue   IP address range in CIDR notation (e.g. "198.51.100.0/24" or "2001:0db8::/32")
+     * @return true if address is in range; else false
+     */
+    public static ExprBooleanValue isAddressInRange(ExprValue addressExprValue, ExprValue rangeExprValue) {
+
+        try {
+            byte[] addressBytes = InetAddresses.forString(addressExprValue.stringValue()).getAddress();
+
+            String[] rangeFields = rangeExprValue.stringValue().split("/");
+            int prefixLengthBytes = Integer.parseInt(rangeFields[1]) / Byte.SIZE;
+            byte[] rangeBytes = Arrays.copyOfRange(InetAddresses.forString(rangeFields[0]).getAddress(), 0, prefixLengthBytes);
+
+            return ExprBooleanValue.of(Arrays.equals(addressBytes, 0, prefixLengthBytes, rangeBytes, 0, prefixLengthBytes));
+        } catch (Exception e) {
+            return ExprBooleanValue.of(false);
```

Comment:
Should this return ExprNullValue instead?

Reply:
Actually these parameters are inputs from the user PPL (and cannot be field references), is that correct? Would it make sense for the parser to resolve this instead? And throw a syntax exception if it is a bad format?

Reply:
I have updated the logic for `cidr(address, range)` (now implemented in `CidrExpression`) so that it throws a semantic error if `range` does not define a valid IP address range in CIDR notation, and returns `ExprNullValue` if `address` is not within the specified range. Let me know what you think.

I looked into using the parser to verify the syntax, but this doesn't seem consistent with the existing approach for other functions. I'm not aware of any other functions that do this? Instead, more "complex" functions seem to have `FunctionExpression` sub-classes, such as `RegexExpression`, that do this sort of "validation" internally, rather than doing so in the parser. Let me know what you think!

Reply:
Update: The error handling behaviour is now based on the Spark implementation. We no longer return `ExprNullValue` under any circumstance: either true/false is returned or an error is thrown.